File: 프로그래머스/Level2/오픈채팅방.py
def solution(record):
    a = []
    obj = {}
    for re in record:
        re = re.split(" ")
        idx = re[1]
        if re[0] == "Enter":
            obj[idx] = re[2]
        elif re[0] == "Change":
            obj[idx] = re[2]
                    
    for rj in record:
        answer = ""
        rj = rj.split(" ")
        idx = rj[1]
        if rj[0] == "Enter":
            answer = obj[idx]+"님이 들어왔습니다."
            a.append(answer)
        elif rj[0] == "Leave":
            answer = obj[idx]+"님이 나갔습니다."
            a.append(answer)
            
    return a

File: 프로그래머스/Level1/오픈채팅방.js
function solution(record) {
    var answer = [];
    var id ={};
    for (var i of record){
        var j = i.split(" ")
        if ( j[0] == 'Enter'){
            id[j[1]] = j[2]
        } else if (j[0] == 'Change'){
            id[j[1]] = j[2]           
        }
    }
    var a =''
    for (var s of record){
        a = ''
        var k = s.split(" ")
        if ( k[0] == 'Enter'){
            a = id[k[1]] + '님이 들어왔습니다.'
            answer.push(a)
        } else if (k[0] == 'Leave'){
            a = id[k[1]] + '님이 나갔습니다.'
            answer.push(a)
        } 
        
    }
    return answer;
}